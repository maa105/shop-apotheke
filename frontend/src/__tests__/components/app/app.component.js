import React from 'react';
import { render } from '@testing-library/react';

import AppComponent from '../../../components/app/app.component';
import LayoutComponent from '../../../components/layout/layout.component';
import { setState } from 'react-redux';

jest.mock('../../../components/layout/layout.component', () => {
  return jest.fn(() => <div>Layout Mock</div>)
});
jest.mock('../../../components/repos/repos-page.component.js', () => {
  return jest.fn(() => <div>Repos Page</div>)
});

jest.mock('react-redux', () => {
  var state = {};
  return {
    setState: (_state) => { state = _state; },
    useSelector: jest.fn((f) => f(state))
  }
});

beforeEach(() => {
  jest.clearAllMocks();
});

it('renders nulls for invalid page', () => {
  setState({ router: { routerState: { page: 'page' } } });
  const component = render(
    <AppComponent />
  );
  expect(component.asFragment()).toMatchSnapshot();
  
  expect(LayoutComponent.mock.calls.length).toEqual(1);
  
  const children = LayoutComponent.mock.calls[0][0].children;

  expect(children).toEqual(null);
});

it('renders repos page', () => {
  setState({ router: { routerState: { page: 'repos' } } });
  const component = render(
    <AppComponent />
  );
  expect(component.asFragment()).toMatchSnapshot();
  
  expect(LayoutComponent.mock.calls.length).toEqual(1);
  
  const appChildren = LayoutComponent.mock.calls[0][0].children;

  expect(appChildren).not.toEqual(null);

  const reposComponent = render(appChildren);
  expect(reposComponent.asFragment()).toMatchSnapshot();
});
