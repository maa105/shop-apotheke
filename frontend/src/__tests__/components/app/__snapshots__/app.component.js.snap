// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders nulls for invalid page 1`] = `
<DocumentFragment>
  <div>
    Layout Mock
  </div>
</DocumentFragment>
`;

exports[`renders repos page 1`] = `
<DocumentFragment>
  <div>
    Layout Mock
  </div>
</DocumentFragment>
`;

exports[`renders repos page 2`] = `
<DocumentFragment>
  <div>
    Repos Page
  </div>
</DocumentFragment>
`;
